lego/apps/notifications/notifications.py

```diff
@@ -16,14 +16,23 @@ def generate_mail(self):
         """
         announcement = self.kwargs["announcement"]
 
+        sender = (
+            announcement.from_group
+            if announcement.from_group
+            else announcement.created_by.full_name
+        )
+
         return self._delay_mail(
             to_email=self.user.email_address,
             context={
                 "name": self.user.full_name,
                 "created_by": announcement.created_by.full_name,
+                "from_group": announcement.from_group.name
+                if announcement.from_group is not None
+                else None,
                 "message": announcement.message,
             },
-            subject=f"Viktig melding fra {announcement.created_by.full_name}",
+            subject=f"Viktig melding fra {sender}",
```

Comment:
I would rewrite this a bit to make it more readable, but functionality is good.   
If you want to keep it as is, you should change line 20 to `announcement.from_group.name` so that you don't depend on `AbakusGroup.__str__`.
```
from_group = None
if announcement.from_group:
  from_group = announcement.from_group.name

sender = announcement.created_by.full_name
if from_group:
  sender = from_group

#########
#########   context={
               "from_group": from_group,
#########

```